pkg/abstractions/common/instance.go

```diff
@@ -23,7 +24,9 @@ type AutoscaledInstanceState struct {
 	RunningContainers  int
 	PendingContainers  int
 	StoppingContainers int
-	FailedContainers   int
+
+	// TODO: We can potentially store the strings for RUNNING, PENDING, STOPPING, containers as well
+	FailedContainers []string
```

Comment:
we already have the ability to GetActiveContainers by stub right? this comment may be redundant, although I see your point

Reply:
i had to add the GetActiveContainersByStubId. It replaces the GetActiveContainer**Count**ByStubId function